### PR TITLE
Cross product implementation

### PR DESCRIPTION
**Pull request by V3RGANz**
It can be useful in work with vectors

### CODE REVIEW

File: src/libgeodecomp/geometry/floatcoord.h [hunk 1]
```diff
@@ -674,6 +674,23 @@ class FloatCoord<3>
         return *this;
     }
 
+    template<template<int> class OTHER_COORD>
+    inline
+    FloatCoord<3> operator&(const OTHER_COORD<3>& a) const
```
[the remark this answers is not in the document]
Comment (V3RGANz) [reply src/libgeodecomp/geometry/floatcoord.h]: Ok, make sense, I did this for reasons of concision

[the remark this answers is not in the document]
Comment (V3RGANz) [reply src/libgeodecomp/geometry/floatcoord.h]: @gentryx what do you think about function crossProduct out of class, and method crossProduct, that apply changes to calling object?

[the remark this answers is not in the document]
Comment (gentryx) [reply src/libgeodecomp/geometry/floatcoord.h]: Sounds good to me!

**Comment by V3RGANz** [comment]
find out it's already implemented, so better to leave current implementation for reasons of backward compatibility

https://github.com/gentryx/libgeodecomp/blob/master/src/libgeodecomp/geometry/floatcoord.h#L758

**Pull request closed without merging**